### PR TITLE
Added nullness annotations to 'ScriptExtensionProvider'

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added nullness annotations to `ScriptExtensionProvider`

My pending review for #1264 ... and more.

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptExtensionProvider.java [hunk 1]
```diff
@@ -15,46 +15,51 @@
...
+    Collection<String> getTypes();
 
     /**
      * This method should return an Object of the given type. Note: get can be called multiple times in the scripts use
      * caching where appropriate.
      *
      * @param scriptIdentifier the identifier of the script that requests the given type
      * @param type the type that is requested (must be part of the collection returned by the {@code #getTypes()} method
-     * @return the requested type (non-null)
+     * @return the requested type or null
```
Comment (cweitkamp) [inline bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptExtensionProvider.java:57]: I am not sure if this is a valid change but as all existing implementations do not take care about throwing IAEs I changed it ...

File: bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java [hunk 2]
```diff
@@ -57,58 +59,61 @@
...
-    private RuleRegistry ruleRegistry;
-    private ScriptedRuleProvider ruleProvider;
-    private ScriptedCustomModuleHandlerFactory scriptedCustomModuleHandlerFactory;
-    private ScriptedCustomModuleTypeProvider scriptedCustomModuleTypeProvider;
-    private ScriptedPrivateModuleHandlerFactory scriptedPrivateModuleHandlerFactory;
+    private @NonNullByDefault({}) RuleRegistry ruleRegistry;
+    private @NonNullByDefault({}) ScriptedRuleProvider ruleProvider;
+    private @NonNullByDefault({}) ScriptedCustomModuleHandlerFactory scriptedCustomModuleHandlerFactory;
+    private @NonNullByDefault({}) ScriptedCustomModuleTypeProvider scriptedCustomModuleTypeProvider;
+    private @NonNullByDefault({}) ScriptedPrivateModuleHandlerFactory scriptedPrivateModuleHandlerFactory;
```
Comment (wborn) [inline bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java:79]: They seem to be mandatory so I think constructor injection would make sense.

File: bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptExtensionProvider.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptExtensionProvider.java:57]: The changes makes sense to me.

File: bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java (hunk 2 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java]: Done.

**Pull request merged**